server/src/main/java/org/elasticsearch/action/bulk/TransportShardBulkAction.java

```diff
@@ -269,8 +272,24 @@ static boolean executeBulkItemRequest(BulkPrimaryExecutionContext context, Updat
                 request.ifSeqNo(), request.ifPrimaryTerm());
         } else {
             final IndexRequest request = context.getRequestToExecute();
+            String id = request.id();
+            IndexSettings indexSettings = context.getPrimary().indexSettings();
+            if (indexSettings != null) {
+                int maxDocIdLen = context.getPrimary().indexSettings().getMaxDocIdLength();
+                if (id != null &&
```

Comment:
I think we expect id to be non-null at this time, i.e., we should rather assert that than make it part of the condition.